### PR TITLE
Update wallet events

### LINKED ISSUE #1708: Update wallet events
- [x] Remove accountIndex from event
- [x] Remove irrelevant events if necessary
- [x] Add new events if necessary
- [x] ?

### PR DESCRIPTION
**Pull request by kwek20**
# Description of change

- Removed ConsolidationRequired event
- Removed account index Event (now always a WalletEvent)

## Links to any relevant issues

fixes #1708

### CODE REVIEW

File: bindings/python/iota_sdk/types/event.py [hunk 1]
```diff
@@ -8,16 +8,14 @@ class WalletEventType(IntEnum):
...
-        ConsolidationRequired (0): Consolidation is required.
-        LedgerAddressGeneration (1): Nano Ledger has generated an address.
-        NewOutput (2): A new output was created.
-        SpentOutput (3): An output was spent.
-        TransactionInclusion (4): A transaction was included into the ledger.
-        TransactionProgress (5): A progress update while submitting a transaction.
+        LedgerAddressGeneration (0): Nano Ledger has generated an address.
+        NewOutput (1): A new output was created.
+        SpentOutput (2): An output was spent.
+        TransactionInclusion (3): A transaction was included into the ledger.
```
Comment (Thoralf-M) [inline bindings/python/iota_sdk/types/event.py:14]: Should this be finalized now? Or do we maybe need confirmed and finalized?

File: bindings/python/examples/exchange/4_listen_events.py [hunk 2]
```diff
@@ -28,8 +28,7 @@
 def callback(event):
     """Callback function for the event listener"""
     event_dict = json.loads(event)
-    print('AccountIndex:', event_dict["accountIndex"])
-    print('Event:', event_dict["event"])
+    print('Event:', event)
```
Comment (DaughterOfMars) [inline bindings/python/examples/exchange/4_listen_events.py:31]: Should this still use event_dict?

File: sdk/src/wallet/core/operations/address_generation.rs [hunk 3]
```diff
@@ -36,41 +36,34 @@ impl Wallet {
                 // needs to have it visible on the computer first, so we need to generate it without the
                 // prompt first
                 let options = options.into();
+                #[cfg(feature = "events")]
```
Comment (DaughterOfMars) [inline sdk/src/wallet/core/operations/address_generation.rs:39]: So this emits before generating now?

Comment (kwek20) [reply sdk/src/wallet/core/operations/address_generation.rs:39]: it always did. https://github.com/iotaledger/iota-sdk/blob/develop/sdk/src/wallet/core/operations/address_generation.rs#L62

Comment (DaughterOfMars) [reply sdk/src/wallet/core/operations/address_generation.rs:39]: Ah, diffs are hard 😆

File: bindings/python/iota_sdk/types/event.py (hunk 1 above)
Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/event.py:14]: Let's keep it like this for now, added a comment to https://github.com/iotaledger/iota-sdk/issues/1814

**Review by Thoralf-M — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**